### ISSUE TITLE
value-list: Use `focusout` event instead of `blur`

### ISSUE THREAD

**Issue by anveshmekala** [Esri/calcite-design-system#6526]
### Description

Currently, we are using `blur` event in value-list-item to identify if the focus has shifted from `dragHandle`  and reset the screen reader instructions. Ideally, we should be using `focusout` event which bubbles to the parent `value-list` rather than the `blur` event.  

When the user tabs out of `dragHandle` and the focus shifts to area with heading & description, `focusout` event is not emitted. From the initial debugging , it looks like the usage of `<label/>` which displays heading & description is causing the issue. 

### Proposed Advantages

- with use of `focusout` event we don't need to add any handler at the child component level
- we can leverage existing `focusout` event handler at the parent.
- `focusout` event bubbles where as `blur` doesn't.

### Which Component

calcite-value-list-item

### Relevant Info

Identified when working on #6402

**Comment by geospatialem** [comment]
Closing with the deprecation of `value-list`.
